src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,138 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.AlreadyRegisteredException;
+import core.basesyntax.exceptions.ValidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final long INITIAL_TEST_ID = 0L;
+    private static final int INITIAL_TEST_AGE = 18;
+    private static final int TEST_YOUNG_AGE = 5;
+    private static final String INITIAL_TEST_LOGIN = "Stray228";
+    private static final String TEST_SHORT_LOGIN = "Cat";
+    private static final String TEST_WRONG_LOGIN = "C%)( t";
+    private static final String INITIAL_TEST_PASSWORD = "123456";
+    private static final String TEST_SHORT_PASSWORD = "1234";
+    private static final String TEST_WRONG_PASSWORD = "$@*T@G %@*)%)&H";
+    private static final RegistrationServiceImpl registrationService
+            = new RegistrationServiceImpl();
+    private StorageDaoImpl storageDao = new StorageDaoImpl();

```

Comment:
```suggestion
    private static final RegistrationService registrationService
            = new RegistrationServiceImpl();
    private StorageDao storageDao = new StorageDaoImpl();
```
use abstract types for variable types